cmd/ddev/cmd/auth-ssh.go

```diff
@@ -68,20 +80,113 @@ var AuthSSHCommand = &cobra.Command{
+			// If it has the same name, change it to avoid conflicts
+			// This can happen if you have symlinks to the same key
+			if _, exists := addedKeys[filename]; exists {
+				filename = fmt.Sprintf("%s_%d", filename, i)
+			}
+			addedKeys[filename] = struct{}{}
+			keyPath = util.WindowsPathToCygwinPath(keyPath)
+			mounts = append(mounts, "--mount=type=bind,src="+keyPath+",dst=/tmp/sshtmp/"+filename)
+		}
+
+		dockerCmd := []string{"run", "-it", "--rm", "--volumes-from=" + ddevapp.SSHAuthName, "--user=" + uidStr, "--entrypoint="}
+		dockerCmd = append(dockerCmd, mounts...)
+		dockerCmd = append(dockerCmd, docker.GetSSHAuthImage()+"-built", "bash", "-c", `cp -r /tmp/sshtmp ~/.ssh && chmod -R go-rwx ~/.ssh && cd ~/.ssh && grep -l '^-----BEGIN .* PRIVATE KEY-----' * | xargs -d '\n' ssh-add`)
 
 		err = exec.RunInteractiveCommand("docker", dockerCmd)
 
 		if err != nil {
-			util.Failed("Docker command 'docker %v' failed: %v", dockerCmd, err)
+			helpMessage := ""
+			// Add more helpful message to the obscure error from Docker
+			// Can be triggered if the key is in /tmp on macOS
+			if strings.Contains(err.Error(), "bind source path does not exist") {
+				helpMessage = "\n\nThe specified SSH key path is not shared with your Docker provider."
+			}
+			util.Failed("Docker command 'docker %v' failed: %v %v", echoDockerCmd(dockerCmd), err, helpMessage)
 		}
 	},
 }
 
+// getSSHKeyPaths returns an array of full paths to SSH keys
+// with checks to ensure they are valid.
+func getSSHKeyPaths(sshKeyPathArray []string, acceptsDirsOnly bool, acceptsFilesOnly bool) []string {
+	var files []string
+	for _, sshKeyPath := range sshKeyPathArray {
+		if !filepath.IsAbs(sshKeyPath) {
+			cwd, err := os.Getwd()
+			if err != nil {
+				util.Failed("Failed to get current working directory: %v", err)
+			}
+			fullPath, err := filepath.Abs(filepath.Join(cwd, sshKeyPath))
+			if err != nil {
+				util.Failed("Failed to derive absolute path for SSH key path %s: %v", sshKeyPath, err)
+			}
+			sshKeyPath = fullPath
+		}
+		fi, err := os.Stat(sshKeyPath)
+		if os.IsNotExist(err) {
+			util.Failed("The SSH key path %s was not found", sshKeyPath)
+		}
+		if err != nil {
+			util.Failed("Failed to check status of SSH key path %s: %v", sshKeyPath, err)
+		}
+		if !fi.IsDir() {
+			if acceptsDirsOnly {
+				util.Failed("SSH key path %s is not a directory", sshKeyPath)
+			}
+			files = append(files, sshKeyPath)
+		} else {
+			if acceptsFilesOnly {
+				util.Failed("SSH key path %s is not a file", sshKeyPath)
+			}
+			files, err = fileutil.ListFilesInDirFullPath(sshKeyPath, true)
+			if err != nil {
+				util.Failed("Failed to list files in %s: %v", sshKeyPath, err)
+			}
+		}
+	}
+	return files
+}
+
+// fileIsPrivateKey checks if a file is readable and that it is a private key.
+// Regex isn't used here because files can be huge.
+// The full check if it's really a private key is done with grep -l '^-----BEGIN .* PRIVATE KEY-----'.
```

Comment:
This comment is probably wrong now.

Reply:
I didn't remove `grep -l '^-----BEGIN .* PRIVATE KEY-----'`, it's still there inside the container.

I check only for `-----BEGIN` here, so it's faster.